src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -655,6 +656,15 @@ private static void fixupLegs(List<Leg> legs, State[][] legsStates) {
             String alightRule = null;
 
             for (int j = 1; j < legsStates[i].length; j++) {
+                if (legsStates[i][j].getBackEdge() instanceof TimedTransferEdge) {
```

Comment:
Suggestion: extract a variable for `legsStates[i][j].getBackEdge()`

Reply:
Good idea! Implemented in https://github.com/ibi-group/OpenTripPlanner/pull/52/commits/13c15af68dca85e5bbf53e61eaf1e473b440fd8c.